scripts/ci/automation_full_test.py

```diff
@@ -279,13 +279,13 @@ def run_instance_modules(self, instance_modules):
             azdev_test_result_fp = os.path.join(azdev_test_result_dir, f"test_results_{instance_idx}.serial.xml")
             cmd = ['azdev', 'test', '--no-exitfirst', '--verbose', '--series'] + serial_tests + \
                   ['--profile', f'{profile}', '--xml-path', azdev_test_result_fp, '--pytest-args', '"--durations=10"']
-            error_flag = process_test(cmd, azdev_test_result_fp, live_rerun=fix_failure_tests)
+            serial_error_flag = process_test(cmd, azdev_test_result_fp, live_rerun=fix_failure_tests)
         if parallel_tests:
             azdev_test_result_fp = os.path.join(azdev_test_result_dir, f"test_results_{instance_idx}.parallel.xml")
             cmd = ['azdev', 'test', '--no-exitfirst', '--verbose'] + parallel_tests + \
                   ['--profile', f'{profile}', '--xml-path', azdev_test_result_fp, '--pytest-args', '"--durations=10"']
-            error_flag = process_test(cmd, azdev_test_result_fp, live_rerun=fix_failure_tests)
-        return error_flag
+            parallel_error_flag = process_test(cmd, azdev_test_result_fp, live_rerun=fix_failure_tests)
+        return serial_error_flag or parallel_error_flag
```

Comment:
The `Automation Full Test` will fail if returned `error_flag` is `True`.

In the previous implementation, `parallel_error_flag` overwrites `serial_error_flag`. When `parallel_error_flag=False` and `serial_error_flag=True`, the overall error_flag should be `True` but turns out to be `False`